tests/python/gaia-ui-tests/gaiatest/gaia_test.py

```diff
@@ -927,6 +935,37 @@ def push_resource(self, filename, remote_path=None, count=1):
     def resource(self, filename):
         return os.path.abspath(os.path.join(os.path.dirname(__file__), 'resources', filename))
 
+    def modify_settings(self, settings):
```

Comment:
I would rename this function to make the intent a bit clearer. Reading, for the first time, a function called `modify_settings` which just returns the argument, is really disturbing.

If I understand the previous comments and the unit test correctly, what we want is to have the ability to force a setting at a class level. This function needs to be inherited and should be used only within `GaiaTestCase` or one of its child. 

What do you think of `_override_default_settings(self, settings)`?